compute_tools/src/http/routes/prewarm_lfc.rs

```diff
@@ -0,0 +1,188 @@
+use crate::compute::ComputeNode;
+use anyhow::Context;
+use async_compression::tokio::bufread::{ZstdDecoder, ZstdEncoder};
+use axum::response::{IntoResponse, Response};
+use axum::{Json, http::StatusCode};
+use reqwest::Client;
+use serde::Serialize;
+use std::sync::Arc;
+use tokio::{io::AsyncReadExt, spawn};
+use tracing::{error, info};
+
+type State = axum::extract::State<Arc<ComputeNode>>;
+const KEY: &str = "lfc_state";
+
+pub struct Parts {
+    uri: String,
+    token: String,
+}
+
+impl TryFrom<&Arc<ComputeNode>> for Parts {
+    type Error = &'static str;
+    fn try_from(state: &Arc<ComputeNode>) -> Result<Self, Self::Error> {
+        let state = state.state.lock().unwrap();
+        let Some(pspec) = state.pspec.as_ref() else {
+            return Err("pspec is not present");
+        };
+
+        let Some(ref endpoint_id) = pspec.spec.endpoint_id else {
+            return Err("pspec.endpoint_id missing");
+        };
+        let Some(ref base_uri) = pspec.endpoint_storage_addr else {
+            return Err("pspec.endpoint_storage_addr missing");
+        };
+        let tenant_id = pspec.tenant_id;
+        let timeline_id = pspec.timeline_id;
+
+        let uri = format!("{base_uri}/{tenant_id}/{timeline_id}/{endpoint_id}/{KEY}");
+        let Some(ref token) = pspec.endpoint_storage_token else {
+            return Err("pspec.endpoint_storage_token missing");
+        };
+        let token = token.clone();
+        Ok(Parts { uri, token })
+    }
+}
+
+impl axum::extract::FromRequestParts<Arc<ComputeNode>> for Parts {
+    type Rejection = Response;
+    async fn from_request_parts(
+        _: &mut http::request::Parts,
+        state: &Arc<ComputeNode>,
+    ) -> Result<Self, Self::Rejection> {
+        state.try_into().map_err(|e| {
+            error!("{e}");
+            StatusCode::BAD_REQUEST.into_response()
+        })
+    }
+}
+
+pub(in crate::http) async fn prewarm_lfc_offload(parts: Parts, state: State) {
+    crate::metrics::LFC_PREWARM_OFFLOAD_REQUESTS.inc();
+    spawn(async move { prewarm_lfc_offload_impl(parts, state).await });
+}
```

Comment:
We will probably want a way of checking if the offload is done in order to know when to do the cut-over.

Reply:
Added /prewarm_lfc_offload_status